### PR TITLE
Use SDK CommunicationIdentifier types and hide internal CommunicationIdentifierModel

### PR DESCRIPTION
**Pull request by DominikMe**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/communication/Azure.Communication.MediaComposition/api/Azure.Communication.MediaComposition.netstandard2.0.cs [hunk 1]
```diff
@@ -31,25 +31,6 @@ public partial class AutoGridLayout : Azure.Communication.MediaComposition.Model
...
-        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
-        public override bool Equals(object obj) { throw null; }
-        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
-        public override int GetHashCode() { throw null; }
-        public static bool operator ==(Azure.Communication.MediaComposition.CommunicationCloudEnvironmentModel left, Azure.Communication.MediaComposition.CommunicationCloudEnvironmentModel right) { throw null; }
-        public static implicit operator Azure.Communication.MediaComposition.CommunicationCloudEnvironmentModel (string value) { throw null; }
-        public static bool operator !=(Azure.Communication.MediaComposition.CommunicationCloudEnvironmentModel left, Azure.Communication.MediaComposition.CommunicationCloudEnvironmentModel right) { throw null; }
-        public override string ToString() { throw null; }
-    }
     public partial class CommunicationError
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.MediaComposition/api/Azure.Communication.MediaComposition.netstandard2.0.cs:34]: We should make this internal as well

File: sdk/communication/Azure.Communication.MediaComposition/src/Models/ParticipantInput.cs [hunk 2]
```diff
@@ -9,5 +10,33 @@ namespace Azure.Communication.MediaComposition.Models
     [CodeGenModel("Participant")]
     public partial class ParticipantInput
     {
+        [CodeGenMember("Id")]
+        internal CommunicationIdentifierModel ServiceId { get; set; }
+
+        /// <summary> The CommunicationIdentifier that identifies the participant.</summary>
+        public CommunicationIdentifier Id => CommunicationIdentifierSerializer.Deserialize(ServiceId);
+
+        internal ParticipantInput(CommunicationIdentifierModel serviceId, string call)
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.MediaComposition/src/Models/ParticipantInput.cs:19]: This is copying over the generated constructor, which isn't ideal if the swagger changes significantly (like adding a new property). This was the most straightforward I could come up with to make the constructor internal, looking at instructions: https://github.com/Azure/autorest.csharp#replace-any-generated-member

Maybe there is a better way.

File: sdk/communication/Azure.Communication.MediaComposition/src/Models/ParticipantInput.cs [hunk 3]
```diff
@@ -9,5 +10,33 @@ namespace Azure.Communication.MediaComposition.Models
     [CodeGenModel("Participant")]
     public partial class ParticipantInput
     {
+        [CodeGenMember("Id")]
+        internal CommunicationIdentifierModel ServiceId { get; set; }
+
+        /// <summary> The CommunicationIdentifier that identifies the participant.</summary>
+        public CommunicationIdentifier Id => CommunicationIdentifierSerializer.Deserialize(ServiceId);
```
Comment (DominikMe) [inline sdk/communication/Azure.Communication.MediaComposition/src/Models/ParticipantInput.cs:17]: Keeping the CommunicationIdentifierModel type as the source of truth was the easiest fix, but it adds a small runtime cost whenever the property gets accessed. I guess I can make this a private setter and set it in the constructor.

File: sdk/communication/Azure.Communication.MediaComposition/api/Azure.Communication.MediaComposition.netstandard2.0.cs (hunk 1 above)
Comment (minnieliu) [reply sdk/communication/Azure.Communication.MediaComposition/api/Azure.Communication.MediaComposition.netstandard2.0.cs]: Done :)

**Review by minnieliu — APPROVED** [review]

**Pull request merged**